src/pages/Register/Register.tsx

```diff
@@ -0,0 +1,198 @@
+import React from 'react';
+import { useForm, Controller } from 'react-hook-form';
+import { useRegisterQuery } from './api/useRegisterQuery';
+import { TextField, Button, Box, Grid, Typography } from '@mui/material';
+
+type FormData = {
+  username: string;
+  name: string;
+  surname: string;
+  email: string;
+  password: string;
+  repeat_password: string;
+};
+
+export const Register = () => {
+  const { mutate: postData, error, isLoading } = useRegisterQuery();
```

Comment:
`{ mutate: registerUser }`

you can also export it as `registerUser` and then in code do `registerUser.mutate` or `registerUser.mutateAsync`